### PR TITLE
add index on form_id

### PR DESCRIPTION
**Pull request by mkangia**
Similar to https://github.com/dimagi/commcare-hq/pull/24392/files
Needed for the same ticket

Some stats
An explain on ICDS
```
commcarehq_p1=> EXPLAIN DELETE FROM form_processor_ledgertransaction WHERE form_id = '123';
                                          QUERY PLAN
----------------------------------------------------------------------------------------------
 Delete on form_processor_ledgertransaction  (cost=0.00..1393220.93 rows=234 width=6)
   ->  Seq Scan on form_processor_ledgertransaction  (cost=0.00..1393220.93 rows=234 width=6)
         Filter: ((form_id)::text = '123'::text)
(3 rows)
```

An explain on local system after adding index
```
Delete on form_processor_ledgertransaction  (cost=0.14..8.16 rows=1 width=6)"
"  ->  Index Scan using fp_ledgtrans_formid_0fe795_idx on form_processor_ledgertransaction  (cost=0.14..8.16 rows=1 width=6)"
"        Index Cond: ((form_id)::text = '123'::text)
```

### CODE REVIEW

File: corehq/form_processor/migrations/0091_auto_20190603_2023.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.20 on 2019-06-03 20:23
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+TABLE_NAME = 'form_processor_ledgertransaction'
+INDEX_NAME = 'fp_ledgtrans_formid_0fe795_idx'
```
Comment (mkangia) [inline corehq/form_processor/migrations/0091_auto_20190603_2023.py]: did we change something about length of index names?
i get an error for index name to be less than 30 characters, so i have to make custom one with short forms.

Comment (snopoke) [reply corehq/form_processor/migrations/0091_auto_20190603_2023.py]: not that I'm aware of, you should use whatever index name is suggested by Django

Comment (mkangia) [reply corehq/form_processor/migrations/0091_auto_20190603_2023.py]: I did that and the migration failed to run on my system. i have not set anything differently on my local PG.

Comment (mkangia) [reply corehq/form_processor/migrations/0091_auto_20190603_2023.py]: updated to name as suggested by django

**Review by snopoke — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Comment by mkangia** [comment]
deployed this on prod india and tcl, however i get different output for the explain now
```
prod
PostgreSQL 9.6.6 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.3 20140911 (Red Hat 4.8.3-9), 64-bit

commcarehq_p1=> EXPLAIN DELETE FROM form_processor_ledgertransaction WHERE form_id = '123';
                                             QUERY PLAN
-----------------------------------------------------------------------------------------------------
 Delete on form_processor_ledgertransaction  (cost=9.86..672.53 rows=168 width=6)
   ->  Bitmap Heap Scan on form_processor_ledgertransaction  (cost=9.86..672.53 rows=168 width=6)
         Recheck Cond: ((form_id)::text = '123'::text)
         ->  Bitmap Index Scan on form_proces_form_id_86572d_idx  (cost=0.00..9.82 rows=168 width=0)
               Index Cond: ((form_id)::text = '123'::text)
(5 rows)


ICDS 
PostgreSQL 9.6.10 on x86_64-pc-linux-gnu (Ubuntu 9.6.10-1.pgdg14.04+1), compiled by gcc (Ubuntu 4.8.4-2ubuntu1~14.04.4) 4.8.4, 64-bit
 Delete on form_processor_ledgertransaction  (cost=0.56..972.71 rows=240 width=6)
   ->  Index Scan using form_proces_form_id_86572d_idx on form_processor_ledgertransaction  (cost=0.56..972.71 rows=240 width=6)
         Index Cond: ((form_id)::text = '123'::text)
(3 rows)
```
which looks like ICDS would benefit from it but others wont?

@snopoke @millerdev  thoughts?

**Comment by millerdev** [comment]
I agree, it looks like the index is good for ICDS. Not sure what to make of other envs though. I'm guessing it's not hurting them. Right?

**Comment by mkangia** [comment]
@millerdev i am not sure how frequently form archival is used on other envs(prod mainly). 
I was curious if this was just a different output coz of different versions. In the end i am hoping for an improvement on all envs, if they use form archival.
Do you think i should check the count of ledger transactions on prod?

**Comment by millerdev** [comment]
I doubt it's different output because of different versions. I'd expect the differences are because the query planner is picking a different plan due to different data in each env. I'm not sure how checking the count would help.

**Comment by mkangia** [comment]
> different data in each env

@millerdev what do you mean by different data?
count was just do see if due to less/more number of records it is using a different plan.

**Comment by millerdev** [comment]
> what do you mean by different data?

I mean the contents of the tables are different in each cluster, and that is one piece of information used by the query planner to decide how to execute the query. So it sounds like we're thinking along the same lines here.

**Pull request merged**